### DISCUSSION TITLE
Pulling PPI data with sourcePPI()

### DISCUSSION THREAD

**Discussion by LPotter21** [netZoo/netZooR#278] [Q&A] [answered]
So I am digging into the documentation to get a PANDA analysis up and running in R and was working to determine how best to pull the priors info for human PPI data. I was looking at the sourcePPI() command as a possible way to create that prior network based on other discussions and the documentation here on GitHub.
However, when looking into the results pulled from STRING for a sample list of proteins, all the scores are "1". I looked at the same list of proteins on the STRING website and the scores vary for the different interactions. Looking at the source code for the sourcePPI() command I find this at the end:
```
# assign "score"column  to "1"
  PPI$score <- "1"
  return(PPI)
```

Is this intentional, and if so, why would you want to manually set the scores instead of using the real scores calculated by STRING?

**Comment by marouenbg** [comment, marked as answer]
Hi @LPotter21 ,

Yes you're right, this was done for a very specific use case where the (unweighted) adjacency matrix was needed.

**Reply by marouenbg** [reply]
I added a note about the function in https://github.com/netZoo/netZooR/pull/279

**Reply by LPotter21** [reply]
Thank you for clearing that up! 

Two other notes that I believe would help others getting started with PANDA analysis and trying to use this command. 
First, it is pulling the full STRING networks, not the physical subnetworks as discussed in the original PANDA paper.
<img width="871" alt="Screen Shot 2023-02-10 at 12 14 03 PM" src="https://user-images.githubusercontent.com/100728178/218166173-97e4e018-44c2-429b-b767-dfe281d504ba.png">

Secondly, the analysis guide [here](https://netzoo.github.io/netZooR/articles/pandaRApplicationinGTExData.html#building-a-panda-regulatory-network) mentions the function as the suggested way to pull PPI data.

> The function source.PPI can be used to source the protein-protein interaction in the STRING database.

Mentioning alternative ways (even just the STRINGdb package) in the guide documentation might be beneficial.

**Reply by marouenbg** [reply]
Thank you for the feedback, for the first comment, I added a mention to clarify that the function does not pull the subnetwork https://github.com/netZoo/netZooR/blob/devel/R/sourcePPI.R#L3
For the second comment, I updated the tutorial here: https://github.com/netZoo/netZooR/pull/280
